include/motor-control/core/motor_driver_config.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <map>
```

Comment:
Sadly this uses `malloc`/`new` under the hood and we do want to avoid that. it's also true that what we're looking for doesn't necessarily fit the concept of a map, a general-purpose associative store; we have a finite number of default-valued options. IMO that's a better fit for a aggregate-initializable struct type anyway, e.g.

```
struct RegisterConfig {
    uint32_t gconf = 0;
    uint32_t gstat = ...
    ....
    uint32_t encm_ctrl = 0;
};
```
which can then be initialized in the can tasks like
```
RegisterConfig{
   .gconf = 0x7002,
   .thigh = ....
   // omit what you don't want to configure and it will take the default value
}```

Reply:
oh we should get rid of this since it's not used anymore